### DISCUSSION TITLE
Is there a way auto delete a route when it not activity?

### DISCUSSION THREAD

**Discussion by simeon49** [Kong/kong#7440] [Help] [answered]
### Summary
In our usage scenario, we will dynamically generate many temporary intranet service nodes as needed(like http://172.16.2.41:8888/api/, http://172.16.2.42:8888/api/, http://172.16.2.43:8888/api/...), and we need to uniformly generate a dynamic external network access address for them

e.g.:
GET  http://www.my-exp.com/dynamically-nodes/41    -> http://172.16.2.41:8888/service/
GET  http://www.my-exp.com/dynamically-nodes/42    -> http://172.16.2.42:8888/service/
GET  http://www.my-exp.com/dynamically-nodes/43    -> http://172.16.2.43:8888/service/
...

I wish the `route` can be auto be deleted when no one vist this route in 1 hour. 

Kong version: 2.4

**Comment by Tieske** [comment, marked as answer]
No easy way to achieve that. A custom plugin probably could by making a request against the management API, but having a dataplane node do this would be a big no-go for me security wise.

**Comment by bungle** [comment]
Could we add `ttl` to routes? Though that would not trigger an event.

**Reply by Tieske** [reply]
No, the CP controls the config, not the DP. Adding a ttl would be a bad architectural decision, how to sync this across a Kong cluster?

If you want this, you can probably use the enterprise vitals data from the CP api to check for calls. If nothing for some time, use the mgt api on the CP to delete the route.

**Reply by dndx** [reply]
+1 on @Tieske , it seems that constantly updating the `ttl` in a scalable way would be very difficult to achieve as well.
